hathorlib/base_transaction.py

```diff
@@ -393,6 +396,24 @@ def is_nft_creation(self) -> bool:
         """Returns True if it's an NFT creation transaction"""
         return False
 
+    def is_standard(self, max_output_script_size: int = settings.MAX_OUTPUT_SCRIPT_SIZE,
+                    allow_non_standard_script: bool = False) -> bool:
+        """Return True is the transaction is standard
+        """
+        is_nft_creation = self.is_nft_creation
+        # First we check if any output script exceeds the maximum script size allowed
+        # Then we must check if all outputs are standard, and we allow a non
+        # standard output only in nft creation transactions
+        for output in self.outputs:
+            if len(output.script) > max_output_script_size:
```

Comment:
Also, it looks like this is outside `is_standard_script` on purpose. But it's not obvious to me why. Wouldn't it be better to just have it there? Maybe if `max_output_script_size` is an `Optional[int]` it can be turned off independently if that is the reason.

Reply:
I think the confusion is that I am separating the checks for script type and script size. The `is_standard_script` method is checking only the script type. Maybe I should move the size check to inside `is_standard_script` and change the parameter `allow_non_standard_script` to `allow_non_standard_script_types`. With that we can have control about allowing any script type and any script size.

What do you think?

Reply:
Looks good now.